### PR TITLE
Adding weddings

### LINKED ISSUE #81: Add wedding functionality
Create weddings
Delete weddings
Assign weddings to contacts
Unassign weddings from contacts
Storage for weddings
UI representation of weddings

### LINKED ISSUE #86: Validation of weddings when adding/editing
Need to create an error for the add and edit commands, check that wedding exists when adding/editing.

### PR DESCRIPTION
**Pull request by dasha3412**
Fixes #81 
Fixes #86 
Adds testing for:
* Storage components
* Assign/Unassign/Create/Delete Wedding Parsers

### CODE REVIEW

File: src/main/java/seedu/address/logic/commands/UnassignWeddingCommand.java [hunk 1]
```diff
@@ -0,0 +1,112 @@
...
+    public static final String COMMAND_WORD = "unassign-wedding";
+    public static final String MESSAGE_REMOVE_WEDDING_SUCCESS = "Removed wedding(s) %1$s from %2$s.";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Removes one or multiple weddings from the person identified "
+            + "by the index number used in the last person listing.\n"
+            + "Parameters: INDEX (must be a positive integer) "
+            + "w/[WEDDING]... (can specify multiple weddings)\n"
+            + "Example: " + COMMAND_WORD + " 1 "
+            + "w/Craig's Wedding w/Wedding April 2025.";
```
[the remark this answers is not in the document]
Comment (dasha3412) [reply src/main/java/seedu/address/logic/commands/UnassignWeddingCommand.java:30]: Oh yes I should specify for that, thanks!

File: src/main/java/seedu/address/model/Model.java [hunk 2]
```diff
@@ -117,7 +119,37 @@ public interface Model {
 
     /**
      * Deletes the given tag.
-     * The tag must exist in the AddressBook.
+     * The tag must exist in the Wedlinker.
      */
     void deleteTag(Tag toDelete);
+
+    /**
+     * Returns true if a tag with the same name as {@code toAdd} exists in the Wedlinker.
```
[the remark this answers is not in the document]
Comment (dasha3412) [reply src/main/java/seedu/address/model/Model.java:127]: Yes, thank you!

File: src/main/java/seedu/address/model/ReadOnlyAddressBook.java [hunk 3]
```diff
@@ -20,4 +21,11 @@ public interface ReadOnlyAddressBook {
      * This list will not contain any duplicate tags.
      */
     ObservableList<Tag> getTagList();
+
+    /**
+     * Returns an unmodifiable view of the tags list.
```
[the remark this answers is not in the document]
Comment (dasha3412) [reply src/main/java/seedu/address/model/ReadOnlyAddressBook.java:26]: 👍

File: src/main/java/seedu/address/model/ReadOnlyAddressBook.java [hunk 4]
```diff
@@ -20,4 +21,11 @@ public interface ReadOnlyAddressBook {
...
+     * This list will not contain any duplicate tags.
```
[the remark this answers is not in the document]
Comment (dasha3412) [reply src/main/java/seedu/address/model/ReadOnlyAddressBook.java:27]: 👍

File: src/main/java/seedu/address/model/person/Person.java [hunk 5]
```diff
@@ -61,6 +64,14 @@ public Set<Tag> getTags() {
         return Collections.unmodifiableSet(tags);
     }
 
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
```
[the remark this answers is not in the document]
Comment (dasha3412) [reply src/main/java/seedu/address/model/person/Person.java:68]: Thank you!

File: src/main/java/seedu/address/model/wedding/UniqueWeddingList.java [hunk 6]
```diff
@@ -0,0 +1,149 @@
...
+        }
+
+        if (!target.isSameWedding(editedWedding) && contains(editedWedding)) {
+            throw new DuplicateWeddingException();
+        }
+
+        internalList.set(index, editedWedding);
+    }
+
+    public void setTag(UniqueWeddingList replacement) {
```
[the remark this answers is not in the document]
Comment (dasha3412) [reply src/main/java/seedu/address/model/wedding/UniqueWeddingList.java:70]: Yes sorry, I think I was referring to another function here and pasted it in but forgot to delete it, thanks for checking!

**Review by riccoljy — APPROVED** [review]
LGTM

**Pull request merged**